### PR TITLE
Authentication in the required components of Docker Compose

### PR DESCRIPTION
**Pull request by bsenel**
This pull request is dedicated to enabling authentication of **PostgreSQL** databases, **Elasticsearch**, and **Kibana**. The updated Docker Compose uses secrets to do so. In addition to the secrets, there are two shell scripts to provide authentication aspect in logpiped and kibana. So, logpiped.sh basically sets the credentials, whereas kibana.sh makes a curl post request to elasticsearch to update kibana user's password.

I propose to use squash and merge if this passes the review.

### CODE REVIEW

File: deploy/secrets/elasticsearch_template.secret [hunk 1]
```diff
@@ -0,0 +1,2 @@
+ELASTIC_USERNAME=elastic
+ELASTIC_PASSWORD=jraSTsk4xaTwsakAs
```
Comment (JustinCappos) [inline deploy/secrets/elasticsearch_template.secret]: Why do we check the passwords into the repo?

File: deploy/secrets/kibana_template.secret [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ELASTICSEARCH_USERNAME=kibana
+ELASTICSEARCH_PASSWORD=a54oxRTsKAs
```
Comment (JustinCappos) [inline deploy/secrets/kibana_template.secret]: Passwords in the repo?

File: deploy/secrets/ledger_template.secret [hunk 3]
```diff
@@ -0,0 +1,4 @@
+POSTGRES_USER=cachecash
+POSTGRES_PASSWORD=Le1LesdX5gErsA
```
Comment (JustinCappos) [inline deploy/secrets/ledger_template.secret]: Passwords in the repo?

File: deploy/secrets/logpiped_template.secret [hunk 4]
```diff
@@ -0,0 +1,4 @@
+POSTGRES_USER=cachecash
+POSTGRES_PASSWORD=tjdTr34a1sx4sa0
```
Comment (JustinCappos) [inline deploy/secrets/logpiped_template.secret]: Passwords in the repo?

File: deploy/secrets/publisher_template.secret [hunk 5]
```diff
@@ -0,0 +1,4 @@
+POSTGRES_USER=cachecash
+POSTGRES_PASSWORD=tjiaEasjx4Gnsa
```
Comment (JustinCappos) [inline deploy/secrets/publisher_template.secret]: Passwords in the repo?

**Review by JustinCappos — COMMENTED** [review]
Why are we listing passwords in the repo?  Shouldn't we have this separate and use Docker Secrets / Vault...

File: deploy/secrets/elasticsearch_template.secret (hunk 1 above)
Comment (bsenel) [reply deploy/secrets/elasticsearch_template.secret]: > Why are we listing passwords in the repo? Shouldn't we have this separate and use Docker Secrets / Vault...

We don't list the passwords in use, we only provide the templates for the users to copy and modify the credentials as they desire. Nobody can build and run the images by skipping this step.

I can update the part of **ELASTIC_PASSWORD=jraSTsk4xaTwsakAs** as **ELASTIC_PASSWORD=\<put a password\>** to make it more understandable. Does it make sense?

**Review by JustinCappos — APPROVED** [review]
LGTM

**Pull request merged**